Post-API.js

```diff
@@ -0,0 +1,35 @@
+const express=require("express");
+const XRegExp=require("xregexp");
+const app=express();
+app.use(express.json());
+
+var data={};
+
+app.post("/post",(req,res)=>{
+
+    data=req.body;
+    REF_replace(data.payload.value);
+    res.send(data.payload);
+
+
+})
+    function REF_replace(dat) {
+      
+          for(i in dat){
+            if(dat[i].valueType=='string'){
+              for(ref in data.referenceData){
+                if(dat[i].value.match(XRegExp(`${ref}`))){
+                       dat[i].value=dat[i].value.replace(XRegExp(`{${ref}}`),data.referenceData[ref]);
+                      }
+                    }
+                  }
+          else{
```

Comment:
bit of indentation please. 
eg: select text, ctrl + k, ctrl + f to format selected code in vscode.